src/test/java/org/junit/tests/assertion/AssertionTest.java

```diff
@@ -717,6 +717,27 @@ public void stringNullAndNullWorksToo() {
         assertEquals("null", null);
     }
 
+    private static class NullToString {
+        @Override
+        public String toString() {
+            return null;
+        }
+    }
+
+    @Test
+    public void nullToString() {
+        try {
+            assertEquals(new NullToString(), new NullToString());
+        } catch (AssertionError e) {
+            assertEquals("expected: org.junit.tests.assertion.AssertionTest$NullToString<null> but "
+                            + "was: org.junit.tests.assertion.AssertionTest$NullToString<null>",
+                    e.getMessage());
+            return;
+        }
+
+        fail("Failed on assertion.");
```

Comment:
@marcphilipp there are different kind of tests in this class:
* some have `@Test(expected = AssertionError.class)`
* some have `return` + `fail()`
* some have only `fail()` inside `try-catch`
* some don't have `fail()` / `return`

As this test is expected to throw an exception and that exception is being checked, I think it's better to make sure an exception was thrown. I agree this variant is too verbose, the best approach would be throwing some new runtime exception from the `try-catch` block.